src/funlist.c

```diff
@@ -464,6 +464,60 @@ FUNCTION(fun_graball)
     free_ansi_string(as);
 }
 
+/* ARGSUSED */
+FUNCTION(fun_arrow)
+{
+  /* iteratively evaluates a list of attributes, with a single argument 
+   * and optional arguments. It passes the result of each result to the 
+   * next attribute as %0, similar to fold, but carries all optional argument
+   * info down each level. Repeat until there are no attributes left in the 
+   * list. Fails if there is no argument.
+   */
+
+  PE_REGS *pe_regs;
+  int funccount, per;
+  char result[BUFFER_LEN];
+  char *list[MAX_SORTSIZE] = {NULL};
+  ufun_attrib ufun;
+  ufun_attrib *ufunList;
+  int n;
+
+  n = list2arr_ansi(list, MAX_SORTSIZE, args[0], ' ', 0);
+  ufunList = mush_calloc(n, sizeof(ufun_attrib), "fun_arrow_array");
+
+  for(int i = 0; i < n; i++) {
+    if (!fetch_ufun_attrib(list[i], executor, &ufunList[i], UFUN_OBJECT))
+      return;
+  }
+
+  freearr(list, n);
+
+  mush_strncpy(result, args[1], sizeof result);
+
+  pe_regs = pe_regs_create(PE_REGS_ARG, "fun_arrow");
+  pe_regs_setenv(pe_regs, 0, result);
+  for(int ai = 2, z = 1; ai < nargs; ai++, z++)
```

Comment:
Why "z" and "ai"? Are they mnemonics for something?

(ai I can guess is arg index. I'm not so sure about why 'z' though)

Reply:
Just avoiding re-use of 'i'. Nothing special going on here.

Reply:
Changed 'z' to 'pei' as in 'pe index'.